src/main/java/org/solid/testharness/utils/DataModelBase.java

```diff
@@ -55,18 +52,21 @@
 
     public enum ConstructMode {
         SHALLOW,
+        INC_PARENTS,
         DEEP,
         LIST;
     }
 
-    private static final String SERVER_GRAPH = "CONSTRUCT {<%s> ?p ?o} WHERE {<%s> ?p ?o}";
-    private static final String SERVER_GRAPH_2 = "CONSTRUCT {<%s> ?p ?o. ?o ?p1 ?o1} " +
-            "WHERE {<%s> ?p ?o. OPTIONAL {?o ?p1 ?o1}}";
+    private static final String SERVER_GRAPH = "CONSTRUCT {?s ?p ?o} WHERE {BIND(<%s> as ?s) ?s ?p ?o}";
+    private static final String SERVER_GRAPH_2 = "CONSTRUCT {?s ?p ?o. ?o ?p1 ?o1} " +
+            "WHERE {BIND(<%s> as ?s) ?s ?p ?o. OPTIONAL {?o ?p1 ?o1}}";
+    private static final String SERVER_GRAPH_PARENT = "CONSTRUCT {?s ?p ?o. ?s1 ?p1 ?s} " +
+            "WHERE {BIND(<%s> as ?s) ?s ?p ?o. OPTIONAL {?s1 ?p1 ?s}}";
```

Comment:
Basically, what the query planner would (hopefully) end up doing in this case would be to first do a `conn.getStatements(%s, null, null)`, and then a `conn.getStatements(null, null, %s)`. But I'd argue it is better to do those yourself, so that there is no  possible query planner  hiccups in the way, there is after all no other reasonable query plan.

Reply:
I was never entirely comfortable with the sparql approach but it was working. You have pushed me to go back to it and do a better job (I appreciate the push) so I'm just working on it now. I can get statements from the connection and load them into a model quite easily as it turns out.

Reply:
I had some fun with functional style Java for this - hope you like the result. No more sparql strings in ReportGenerator, DataModelBase and TestSuiteDescription